### ISSUE TITLE
Consider adding `no_leading_underscores_for_local_identifiers` to recommended

### ISSUE THREAD

**Issue by pq** [dart-lang/lints#58]
[no_leading_underscores_for_local_identifiers](https://dart-lang.github.io/linter/lints/no_leading_underscores_for_local_identifiers.html) would be a good candidate for `recommended` if not `core`.

### Lint Description:

**DON’T** use a leading underscore for identifiers that aren't private. Dart
uses a leading underscore in an identifier to mark members and top-level
declarations as private. This trains users to associate a leading underscore
with one of those kinds of declarations. They see `_` and  think "private".
There is no concept of "private" for local variables or parameters.  When one of 
those has a name that starts with an underscore, it sends a confusing signal to
the reader. To avoid that, don't use leading underscores in those names.

**Exception**: An unused parameter can be named `_`, `__`, `___`, etc.  This is
common practice in callbacks where you are passed a value but you don't need
to use it. Giving it a name that consists solely of underscores is the idiomatic
way to indicate that the value isn't used.

**BAD**
```dart
void print(String _name) {
  var _size = _name.length;
  ...
}
```
**GOOD:**

```dart
void print(String name) {
  var size = name.length;
  ...
}
```

**OK:**

```dart
[1,2,3].map((_) => print('Hello'));
```

---

We might also consider [no_leading_underscores_for_library_prefixes](https://dart-lang.github.io/linter/lints/no_leading_underscores_for_library_prefixes.html).

**Comment by goderbauer** [comment]
Trying this out in flutter is currently blocked on https://github.com/dart-lang/linter/issues/3127, the fix for which hasn't rolled into Flutter yet.

**Comment by goderbauer** [comment]
Lint works fine on Flutter's code base now that https://github.com/dart-lang/linter/issues/3127 has rolled in.

**Comment by pq** [comment]
Great to hear.  Thanks for checking!

**Comment by pq** [comment]
See: https://github.com/dart-lang/linter/issues/3216

**Comment by lrhn** [comment]
Do we have (or want) a "no leading underscores for library names" too?

Like
```dart
library _foo;
```
or
```dart
library foo._bar;
```

Using an underscore doesn't hurt, because no-one actually uses library names for anything anymore, but it might surprise people using `dart:mirrors` that `library _foo` has a name that isn't the symbol `#_foo`, because library names are not affected by privacy.
